resdk/resolwe.py

```diff
@@ -293,16 +293,23 @@ def run(self, slug=None, input={}, descriptor=None,  # pylint: disable=redefined
         else:
             raise ValueError("Unexpected behaviour at get process with slug {}".format(slug))
 
-        # Upload files in basic:file fields
+        # Pre-process inputs
         try:
             for schema, fields in iterate_fields(input, process['input_schema']):
                 field_name = schema['name']
                 field_type = schema['type']
                 field_value = fields[field_name]
 
+                # Wrapp `list:` fields into list if they are not already
```

Comment:
> Exceptions should get raised for wrong inputs.

The inputs are not wrong, they are just not formatted in the right way :).

I agree that exception is better from Orange point of view. But on the other side, we should make it friendly for the users that are not top developers (or are not developers at all). So we must consider both sides.

In this case I don't see a problem wrapping value into list as it is clearly specified that list is required and all other options will fail. So there is no "javascript" magic in my opinion.